src/lib/common/globals.cpp

```diff
@@ -340,9 +340,37 @@ int64_t parse8601(const std::string& s)
              ((*duration == 'H') || (*duration == 'M') || (*duration == 'S')))
     {
       char what = *duration;
+      int  value;
 
       *duration = 0;
-      int value = atoi(start);
+
+      if (what == 'S')  // We support floats for the seconds, but only to round to an integer
+      {
+        //
+        // To round e.g. 4.51 to 5, we split 4.51 in its integer part and its decimal part:
+        //   secs    = 4.51
+        //   intPart = 4
+        //   rest    = 0.51
+        //
+        // After this simple operation is done, we just need to add ONE to 'intPart' if
+        // 'rest' >= 0.5 ('greater than' or just 'greater'?)
+        // and if 'rest' < 0.5, it is just thrown away.
+        //
+        float secs    = atof(start);
```

Comment:
What about of just use the round() function (www.gnu.org/software/libc/manual/html_node/Rounding-Functions.html), e.g:

```
value = (int) round(start);
```

Much shorter (just one statement) and clearer code, in my opinion.